be/src/connector_sink/hive_chunk_sink.cpp

```diff
@@ -57,10 +57,11 @@ StatusOr<ConnectorChunkSink::Futures> HiveChunkSink::add(ChunkPtr chunk) {
     }
 
     // create writer if not found
-    if (_partition_writers[partition] == nullptr) {
+    if (!_partition_writers.contains(partition)) {
         auto path = _partition_column_names.empty() ? _location_provider->get() : _location_provider->get(partition);
-        ASSIGN_OR_RETURN(_partition_writers[partition], _file_writer_factory->create(path));
-        RETURN_IF_ERROR(_partition_writers[partition]->init());
+        ASSIGN_OR_RETURN(auto writer, _file_writer_factory->create(path));
+        RETURN_IF_ERROR(writer->init());
+        _partition_writers[partition] = writer;
```

Comment:
You can use `_partition_writers.emplace(partition, std::move(writer))`

Reply:
get